tests/test_run_ccoverage.py

```diff
@@ -20,11 +24,43 @@
 logging.getLogger("flake8").setLevel(logging.WARNING)
 
 
+def mock_ccov_time():
+    """Overwrite the ccov_time function to return a shorter time to test code coverage.
+
+    Returns:
+        int: Number of seconds to run code coverage
+    """
+    return 3
+
+
+def mock_covreporter_main(argv=None):  # pylint: disable=unused-argument
+    """Overwrite the main function in CovReporter to not submit reports during the test run.
+
+    Args:
+        argv (list): List of parameters to be passed to CovReporter
+
+    Returns:
+        int: Simulates a successful submission exit code
+    """
+    FUNFUZZ_TEST_LOG.info("Simulating a successful submission...")
+    return 0
+
+
 class RunCcoverageTests(unittest.TestCase):
     """"TestCase class for functions in run_ccoverage.py"""
-    @pytest.mark.skip(reason="disable for now until actual use")
-    def test_main(self):
+
+    @staticmethod
+    @pytest.mark.skipif(distro.linux_distribution()[0] == "Ubuntu" and
+                        parse_version(distro.linux_distribution()[1]) < parse_version("16.04"),
+                        reason="Code coverage binary crashes in 14.04 Trusty but works in 16.04 Xenial and up")
+    @pytest.mark.slow
+    def test_main():
         """Run run_ccoverage with test parameters."""
         build_url = "https://build.fuzzing.mozilla.org/builds/jsshell-mc-64-opt-gcov.zip"
-        # run_ccoverage's main method does not actually return anything.
-        self.assertTrue(not funfuzz.run_ccoverage.main(argparse_args=["--url", build_url]))
+
+        monkey = MonkeyPatch()
+        with monkey.context() as monkey_context:
+            monkey_context.setattr(funfuzz.ccoverage.gatherer, "ccov_time", mock_ccov_time)
+            monkey_context.setattr(CovReporter, "main", mock_covreporter_main)
+
+            assert not funfuzz.run_ccoverage.main(argparse_args=["--url", build_url, "--report"])
```

Comment:
You could check that `mock_covreporter_main` is called by capturing the log calls, and check that `argv` is as expected.

Reply:
This was what I was referring to: [test_lithium.py](https://github.com/MozillaSecurity/lithium/blob/eda41d328ecd7092ad858ec13b3c0bf7e6969b7e/tests/test_lithium.py#L425-L437)

Reply:
I solved this by monkeypatching the `report_coverage` and `disable_pool` functions instead.